client/src/components/Login.js

```diff
@@ -66,11 +69,12 @@ const Login = ({ toggleForm }) => {
     e.preventDefault();
     setLoginPending(true);
     try {
-      const responseStatus = await login(loginData);
+      const { responseStatus, user } = await login(loginData);
 
       if (responseStatus === 200) {
+        setUser(user)  // store user in state because user is checked to determine private route access
         setLoggedIn(true);
-        navigate("/");
+        navigate("/", { state:user }); // store user in route state because "/" route will render with the old user state at this point
```

Comment:
QQ here. If we update the state via our context as we do a few lines up from this, does route "/" render old user state data? I haven't really used this feature from react router dom, but it looks like its a way to carry state across routes. If so our user context is already doing this, correct?